### PR TITLE
Persistent state for Overlay widget toggles

### PR DESCRIPTION
**Pull request by WSSDude420**
Keeps previously opened widgets opened after game shuts down when it next launches.

Also organize options a bit better and fix mod being able to draw during first time setup.

Dependency for https://github.com/yamashi/CyberEngineTweaks/pull/754
Continued in #754 (gave it separate file there and some other tweaks)

### CODE REVIEW

File: src/Options.cpp [hunk 1]
```diff
@@ -3,106 +3,206 @@
...
+      {"disable_intro_movies", DisableIntroMovies},
+      {"disable_vignette", DisableVignette},
+      {"disable_boundary_teleport", DisableBoundaryTeleport},
+      {"disable_win7_vsync", DisableWin7Vsync},
+      {"minimap_flicker", MinimapFlicker},
+    };
 }
 
-void Options::Save()
+void PatchesSettings::ResetToDefaults()
```
Comment (yamashi) [inline src/Options.cpp:37]: Could just write *this = {};

File: src/Options.cpp [hunk 2]
```diff
@@ -3,106 +3,206 @@
...
+      {"oversample_vertical", OversampleVertical}
+    };
+}
 
-    config["font_path"] = FontPath;
-    config["font_glyph_ranges"] = FontGlyphRanges;
-    config["font_size_base"] = FontSizeBase;
-    config["font_oversample_horizontal"] = FontOversampleHorizontal;
-    config["font_oversample_vertical"] = FontOversampleVertical;
+void FontSettings::ResetToDefaults()
```
Comment (yamashi) [inline src/Options.cpp:71]: *this = {};

File: src/Options.cpp [hunk 3]
```diff
@@ -3,106 +3,206 @@
...
+    return {
+      {"remove_dead_bindings", RemoveDeadBindings},
+      {"enable_imgui_assertions", EnableImGuiAssertions},
+      {"enable_debug", EnableDebug},
+      {"dump_game_options", DumpGameOptions},
+    };
+}
 
+void DeveloperSettings::ResetToDefaults()
+{
```
Comment (yamashi) [inline src/Options.cpp:105]: *this = {}

**Review by yamashi — CHANGES_REQUESTED** [review]

File: src/Options.cpp (hunk 1 above)
Comment (WSSDude420) [reply src/Options.cpp:37]: did not thought of that 💡

Comment (WSSDude420) [reply src/Options.cpp:37]: changed all of these

**Review by yamashi — APPROVED** [review]

**Pull request merged**